app/models/assignment.rb

```diff
@@ -827,9 +827,11 @@ def is_criteria_mark?(ta_id)
     assign_graders_to_criteria && self.criterion_ta_associations.where(ta_id: ta_id).any?
   end
 
-  def get_num_assigned(ta_id = nil)
+  def get_num_assigned(ta_id = nil, bulk: false)
     if ta_id.nil?
       groupings.size
+    elsif bulk
+      cache_ta_results.dig(ta_id, :total_results)&.size
```

Comment:
You'll need to use an `|| 0` to ensure an integer is always returned (rather than `nil`)